src/kbmod/search/raw_image_eigen.cpp

```diff
@@ -0,0 +1,767 @@
+    image = std::move(img);
+    height = image.rows();
+    width = image.cols();
+    obstime = obs_time;
+  }
+
+
+  RawImageEigen::RawImageEigen(unsigned w, unsigned h, float value, double obs_time)
+    : height(h),
+      width(w),
+      obstime(obs_time) {
+    if (value != 0)
+      image = Image::Constant(height, width, value);
+    else
+      image = Image::Zero(height, width);
+  }
+
+
+  // Copy constructor
+  RawImageEigen::RawImageEigen(const RawImageEigen& old) {
+    width = old.get_width();
+    height = old.get_height();
+    image = old.get_array();
+    obstime = old.get_obstime();
+  }
+
+
+  // Move constructor
+  RawImageEigen::RawImageEigen(RawImageEigen&& source)
+    : width(source.width),
+      height(source.height),
+      obstime(source.obstime),
+      image(std::move(source.image)) {}
+
+
+  // Copy assignment
+  RawImageEigen& RawImageEigen::operator=(const RawImageEigen& source) {
+    width = source.width;
+    height = source.height;
+    image = source.image;
+    obstime = source.obstime;
+    return *this;
+  }
+
+
+  // Move assignment
+  RawImageEigen& RawImageEigen::operator=(RawImageEigen&& source) {
+    if (this != &source) {
+      width = source.width;
+      height = source.height;
+      image = std::move(source.image);
+      obstime = source.obstime;
+    }
+    return *this;
+  }
+
+
+  bool RawImageEigen::isclose(const RawImageEigen& img_b, float atol) const {
+    return image.isApprox(img_b.image, atol);
+  }
+
+
+  auto RawImageEigen::get_interp_neighbors_and_weights(const float x, const float y) const {
+    Point p(x, y);
+
+    // coordinates of nearest neighboring array
+    // top left (tl), top right (tr),
+    // bottom left (bl), bottom right (br)
+    auto [tl, tr, bl, br] = p.nearest_pixel_coords();
+
+    // interpolation weights for each of the 4 neighbors
+    // total normalization should be 1 always (size of array)
+    // float normalization = 1 / ((tr.x - tl.x) * (tl.y - bl.y));
+    float w_tl = (tr.x - x) * (y - bl.y);
+    float w_tr = (x - tl.x) * (y - bl.y);
+    float w_bl = (br.x - x) * (tr.y - y);
+    float w_br = (x - bl.x) * (tl.y - y);
+
+    auto [itl, itr, ibl, ibr] = p.nearest_pixel_idxs();
+
+    struct retval {
+      // gee golly naming is hard...
+      Index top_left, top_right, bot_left, bot_right;
+      float w_tl, w_tr, w_bl, w_br;
+    };
+
+    return retval{itl, itr, ibl, ibr, w_tl, w_tr, w_bl, w_br};
+  }
+
+
+  float RawImageEigen::interpolate2(const float x, const float y) const {
+    Point p(x, y);
+
+    // neighbors and weights for interpolation
+    auto iwn = get_interp_neighbors_and_weights(x, y);
+
+    // I sure hope the compiler optimizes locals away successfully
+    auto tl = iwn.top_left;
+    auto tr = iwn.top_right;
+    auto bl = iwn.bot_left;
+    auto br = iwn.bot_right;
+
+    // dodge masked values (could also have been an expression,
+    // or for loop, not sure about preferred C++ ways or benefits?
```

Comment:
I don't think it can be the expression below because of the need to change the summing (as described below). A loop would work, but I think the if statements are simpler in this case (since there are only 4).

Reply:
Removed and reverted to old behavior.